### PR TITLE
Show total execution time

### PR DESCRIPTION
**Pull request by brunoqueiros**
Hello,

This PR shows the total execution time of all tests. It's a nice information to show.

image

### CODE REVIEW

File: lib/logger.js [hunk 1]
```diff
@@ -64,11 +64,14 @@ x.errors = function (tests) {
 	});
 };
 
-x.report = function (passed, failed, unhandled, uncaught) {
+x.report = function (passed, failed, unhandled, uncaught, startTime) {
+	var duration = new Date() - startTime;
```
Comment (vdemedes) [inline lib/logger.js]: Logger should not calculate total time, it should only write output.

File: api.js [hunk 2]
```diff
@@ -24,6 +24,7 @@ function Api(files, options) {
 	this.failCount = 0;
 	this.fileCount = 0;
 	this.testCount = 0;
+	this.startTime = new Date();
```
Comment (vdemedes) [inline api.js]: `lib/globals.js` and its `now()` method should be used instead of `new Date()`.

File: test/cli.js [hunk 3]
```diff
@@ -69,7 +70,7 @@ test('don\'t display test title if there is only one anonymous test', function (
 
 	execCli(['fixture/es2015.js'], function (err, stdout, stderr) {
 		t.ifError(err);
-		t.is(stderr.trim(), '1 test passed');
+		t.true(stderr.trim().includes('1 test passed'));
```
Comment (vdemedes) [inline test/cli.js]: This functionality should be tested by using API (`api.js`), without executing AVA's CLI.

**Comment by sindresorhus** [comment]
> It's a nice information to show.

We don't add things on the merit of it being "nice to have". Does it have any actual use-case for every test run? I think knowing the test time can be useful once in awhile to make sure your tests aren't getting too slow, but then you can just do `$ time ava`. I don't want to see the time every time I run a test and I don't really see how it's useful either.

**Comment by brunoqueiros** [comment]
In my opinion some users want to see the total execution time, if you don't want show the time for all users could be a good solution have a flag, that way only users which set the flag will see.

**Comment by vdemedes** [comment]
@brunoqueiros If users will need to use a flag (for example, `--show-time`), they might as well just:

```js
$ time ava
```

which will also show total execution time.

**Comment by sindresorhus** [comment]
> In my opinion some users want to see the total execution time

Why?

**Comment by brunoqueiros** [comment]
OK, if you doesn't see that is a useful feature I'll close the PR.

**Pull request closed without merging**